core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -61,9 +61,10 @@
 
   /**
    * Force this sorter to spill when there are this many elements in memory. The default value is
-   * 1024 * 1024 * 1024 / 2 which allows the maximum size of the pointer array to be 8G.
+   * 1024 * 1024 * 1024 / 4 which allows the maximum size of the pointer array(including the
```

Comment:
Instead of explaining things this way, which is what caused the bug originally, how about doing the calculation the opposite way? e.g.

```
private final static long ENTRY_SIZE = Long.BYTES * 4;
private final static long MAX_ARRAY_SIZE = ByteUnit.GiB.toBytes(8L);
public final static long DEFAULT_NUM_ELEMENTS_FOR_SPILL_THRESHOLD = MAX_ARRAY_SIZE / ENTRY_SIZE;
```

Results in the same value, but the code is self-explanatory (assuming `ENTRY_SIZE` is correct).